### DISCUSSION TITLE
Help regarding custom kitten creation

### DISCUSSION THREAD

**Discussion by DarkblooM-IO** [kovidgoyal/kitty#7879] [Q&A]
```python
from typing import List
from kitty.boss import Boss

def main(args: List[str]) -> str:
    # this is the main entry point of the kitten, it will be executed in
    # the overlay window when the kitten is launched
    answer = input('Enter some text: ')
    # whatever this function returns will be available in the
    # handle_result() function
    return answer

def handle_result(args: List[str], answer: str, target_window_id: int, boss: Boss) -> None:
    # get the kitty window into which to paste answer
    w = boss.window_id_map.get(target_window_id)
    if w is not None:
        w.paste_text(answer)
```

In this example program, is `w.paste_text(answer)` supposed to print out `answer` to the standard output? Because if so, it doesn't seem to be working for me.

Enlightenment would be appreciated.

**Comment by kovidgoyal** [comment]
What exactly does "it isnt working" mean?

**Reply by DarkblooM-IO** [reply]
What I mean is, if the `paste_text` function is supposed to print out the argument(s) it is passed, then in this example it is not doing so for me. The "Enter some text" prompt is correctly executed but then nothing happens.

**Reply by kovidgoyal** [reply]
and some prints to debug you will see the output in the stdout of the kitty process which you run from another terminal. Most likely it will be because target_window_id no longer exists.

**Reply by DarkblooM-IO** [reply]
I think the problem is that the `handle_result` function is never called. I tried adding some print statements inside it but they never happen.

**Comment by kovidgoyal** [comment]
You need to run the kitten via  a mapping in kitty.conf for that

**Reply by DarkblooM-IO** [reply]
Oh OK. So if I just want to run it as a command I don't need `handle_result`?
